rcl/src/rcl/rmw_implementation_identifier_check.c

```diff
@@ -131,30 +128,36 @@ INITIALIZER(initialize) {
   if (expected_rmw_impl) {
     const char * actual_rmw_impl_id = rmw_get_implementation_identifier();
     if (!actual_rmw_impl_id) {
-      RCUTILS_LOG_ERROR_NAMED(
-        ROS_PACKAGE_NAME,
+      RCL_SET_ERROR_MSG_WITH_FORMAT_STRING(
         "Error getting RMW implementation identifier / RMW implementation not installed "
         "(expected identifier of '%s'), with error message '%s', exiting with %d.",
         expected_rmw_impl,
         rcl_get_error_string().str,
         RCL_RET_ERROR
       );
-      rcl_reset_error();
```

Comment:
If this was resetting the error before, now you seem to be overriding an error message

Reply:
How do you suggest me addressing this? Using an extra string to save the error msg before resetting it?

The current code uses the previous error string as part of the new error before overriding it. I think it will throw an extra warning at the exit output, but I thought it wasn't that much of a trouble considering that in this scenario the code exited the program

Reply:
> How do you suggest me addressing this? Using an extra string to save the error msg before resetting it?

Exactly. You can use [rcutils_get_error_string](https://github.com/ros2/rcutils/blob/ddd1c1202d3e726baddbfbdafd3034a0c24f313a/include/rcutils/error_handling.h#L258), followed by [rcl_reset_error](https://github.com/ros2/rcutils/blob/ddd1c1202d3e726baddbfbdafd3034a0c24f313a/include/rcutils/error_handling.h#L263), followed by setting the new formatted error string.

Reply:
Done 334cbef